### ISSUE TITLE
half-integers spins

### ISSUE THREAD

**Issue by cassio-marinho** [BlackHolePerturbationToolkit/SpinWeightedSpheroidalHarmonics#22]
Recently I've tried to get the Eigenvalue for half-integer spins using the SWSH package.

Suggested by Marc Casals, I have changed the statement that {s,l,m} is integer. Then it works.

So, maybe the best way would be asking {s,l,m} to be integer or half-integer.

**Comment by nielsw2** [comment]
Do you know of a published table of values for the eigenvalue for half-integer spins that we could compare against?

**Comment by cassio-marinho** [comment]
No. Right now I'm trying to find it. So far, the most that I found was a statement from https://arxiv.org/pdf/1810.00432.pdf 

Teukolsky managed to decouple and separate by variables the linear spin-field perturbations of Kerr black holes. He achieved this for the radiative components of the massless fields of spin (helicity) s= 0(scalar),±1/2(neutrino),±1(electromagnetic) and±2(gravitational). 

Implying that it should word for half integer spin too.

**Comment by barrywardell** [comment]
What values of {s,l,m} are allowed? For example, is it the case that l+m must be an integer? And must they both differ from s by an integer?

**Comment by nielsw2** [comment]
Andrew Coates and Sebstian Völkel wrote to me in October 2020 saying they'd found an issue with the small gamma expansion and the numerical calculation of the eigenvalue for s=1/2. I passed this on to Marc Casals who suggested a simple fix for the series expansion by defining:

\[Alpha][s_, l_, m_] /; l == Abs[m] = 0;

The numerical issue seems to be entirely in the spherical expansion method (Leaver's method still works if a good guess is given). The issue only seems to appear for s=1/2, l=1/2 (this is a valid parameter combination? -- see Barry's question above) in the kHat equation. In this case it is easily fixed by taking the limit for this one special case.

I've created a new branch `halfIntegerSpin` with the proposed fixes. It would be good to check the results and add one or two unit tests for these cases.

It was not clear me if the expansion about gamma = infinity is working in the s=1/2 case.

**Comment by barrywardell** [comment]
This appears to fix the error message. Once we can confirm that the result is correct then we can merge the fix.

**Comment by nielsw2** [comment]
Appendix B of https://arxiv.org/abs/0905.2974 has values we can compare against.

**Comment by Kevin-Cunningham** [comment]
[halfSpinSWSHcheck.zip](https://github.com/BlackHolePerturbationToolkit/SpinWeightedSpheroidalHarmonics/files/11088403/halfSpinSWSHcheck.zip)

> Appendix B of https://arxiv.org/abs/0905.2974 has values we can compare against.

I've checked the values in the appendix against the toolkit eigenvalues. They seem to agree everywhere that they should (after correcting for different conventions etc). The notebook where I made the checks is attached.
